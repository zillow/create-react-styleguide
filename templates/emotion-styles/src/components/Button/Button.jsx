import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

export const StyledButton = styled('button')(props => props.theme.Button);
const Button = props => <StyledButton {...props} />;

Button.propTypes = {
    /** Button content */
    children: PropTypes.node.isRequired,
    /** Button type */
    type: PropTypes.oneOf(['primary', 'secondary']),
    /** Enable outline buttons */
    outline: PropTypes.bool,
};

Button.defaultProps = {
    type: 'primary',
    outline: false,
};

export default Button;
