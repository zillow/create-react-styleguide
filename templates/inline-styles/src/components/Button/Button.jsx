import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.Component {
    static propTypes = {
        /** Button content */
        children: PropTypes.node.isRequired,
        /** Button type */
        type: PropTypes.oneOf(['primary', 'secondary']),
        /** Enable outline buttons */
        outline: PropTypes.bool
    };
    static defaultProps = {
        type: 'primary'
    };

    render() {
        let styles = {
            display: 'inline-block',
            fontWeight: '400',
            textAlign: 'center',
            whiteSpace: 'nowrap',
            verticalAlign: 'middle',
            userSelect: 'none',
            border: '1px solid transparent',
            padding: '.375rem .75rem',
            fontSize: '1rem',
            lineHeight: '1.5',
            borderRadius: '.25rem',
            transition: 'color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out',
            color: '#fff',
            backgroundColor: '#007bff',
            borderColor: '#007bff'
        };

        if (this.props.outline && this.props.type == 'primary') {
            styles = Object.assign(styles, {
                color: '#007bff',
                backgroundColor: 'transparent',
                backgroundImage: 'none',
                borderColor: '#007bff'
            });
        } else if (this.props.outline && this.props.type == 'secondary') {
            styles = Object.assign(styles, {
                color: '#6c757d',
                backgroundColor: 'transparent',
                backgroundImage: 'none',
                borderColor: '#6c757d'
            });
        } else if (this.props.type === 'secondary') {
            styles = Object.assign(styles, {
                backgroundColor: '#6c757d',
                borderColor: '#6c757d'
            });
        }

        return <button style={styles} {...this.props} />;
    }
}
