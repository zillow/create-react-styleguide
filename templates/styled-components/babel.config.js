module.exports = {
    presets: [['babel-preset-zillow', { modules: false }]],
    env: {
        cjs: {
            presets: ['babel-preset-zillow'],
        },
        test: {
            presets: [
                [
                    'babel-preset-zillow',
                    {
                        // Fixes "No styles found on passed component"
                        // https://github.com/styled-components/jest-styled-components/issues/294
                        'styled-components': {
                            displayName: false,
                            ssr: false,
                            namespace: 'sc',
                        },
                    },
                ],
            ],
        },
    },
};
