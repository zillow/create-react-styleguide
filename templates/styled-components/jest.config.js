const { createJestConfig } = require('create-react-styleguide');

module.exports = Object.assign(createJestConfig(), {});
