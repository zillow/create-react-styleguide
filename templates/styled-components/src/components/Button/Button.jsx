import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ButtonMixin } from '../../mixins/ButtonMixin';

export const StyledButton = styled.button`
    ${ButtonMixin}
`;

/**
 * A simple wrapper around a native
 * [`button`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) element.
 */
const Button = React.forwardRef((props, ref) => <StyledButton ref={ref} {...props} />);

Button.displayName = 'Button';

Button.propTypes = {
    /** Button content */
    children: PropTypes.node.isRequired,
    /** Button type */
    variant: PropTypes.oneOf(['primary', 'secondary']),
    /** Enable outline buttons */
    outline: PropTypes.bool,
};

Button.defaultProps = {
    variant: 'primary',
    outline: false,
};

export default Button;
