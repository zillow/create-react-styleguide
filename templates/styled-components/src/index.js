// Components
export { default as Button } from './components/Button/Button';

// Themes
export { default as ThemeBootstrap } from './themes/ThemeBootstrap';
