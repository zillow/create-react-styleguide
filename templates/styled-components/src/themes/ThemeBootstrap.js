const ThemeBootstrap = {
    colors: {
        blue: '#007bff',
        gray: '#6c757d',
        white: '#fff',
    },
};

ThemeBootstrap.Button = {
    primary: {
        backgroundColor: ThemeBootstrap.colors.blue,
        borderColor: ThemeBootstrap.colors.blue,
        color: ThemeBootstrap.colors.white,

        hoverBackgroundColor: '#0069d9',
        hoverBorderColor: '#0062cc',
        hoverColor: ThemeBootstrap.colors.white,
    },
    secondary: {
        backgroundColor: '#6c747d',
        borderColor: ThemeBootstrap.colors.gray,
        color: ThemeBootstrap.colors.white,

        hoverBackgroundColor: '#5a6268',
        hoverBorderColor: '#545b62',
        hoverColor: ThemeBootstrap.colors.white,
    },
    primaryOutline: {
        backgroundColor: 'transparent',
        borderColor: ThemeBootstrap.colors.blue,
        color: ThemeBootstrap.colors.blue,

        hoverBackgroundColor: ThemeBootstrap.colors.blue,
        hoverBorderColor: ThemeBootstrap.colors.blue,
        hoverColor: ThemeBootstrap.colors.white,
    },
    secondaryOutline: {
        backgroundColor: 'transparent',
        borderColor: ThemeBootstrap.colors.gray,
        color: ThemeBootstrap.colors.gray,

        hoverBackgroundColor: ThemeBootstrap.colors.gray,
        hoverBorderColor: ThemeBootstrap.colors.gray,
        hoverColor: ThemeBootstrap.colors.white,
    },
};

export default ThemeBootstrap;
