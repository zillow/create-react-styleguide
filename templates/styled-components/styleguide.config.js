const path = require('path');
const { createStyleguideConfig } = require('create-react-styleguide');

module.exports = createStyleguideConfig({
    /* your own config shallowly merged */
    sections: [
        {
            name: 'Introduction',
            content: 'docs/introduction.md',
        },
    ],
    styleguideComponents: {
        Wrapper: path.join(__dirname, 'src/styleguidist/ThemeWrapper'),
    },
});
