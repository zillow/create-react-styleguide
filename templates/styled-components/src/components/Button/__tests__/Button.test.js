import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Button from '../Button';
import ThemeBootstrap from '../../../themes/ThemeBootstrap';

describe('Button', () => {
    it('renders a primary button', () => {
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={ThemeBootstrap}>
                <Button variant="primary">Click Me!</Button>
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule(
            'background-color',
            ThemeBootstrap.Button.primary.backgroundColor
        );
        expect(tree).toMatchSnapshot();
    });

    it('renders a secondary button', () => {
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={ThemeBootstrap}>
                <Button variant="secondary">Click Me!</Button>
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule(
            'background-color',
            ThemeBootstrap.Button.secondary.backgroundColor
        );
        expect(tree).toMatchSnapshot();
    });

    it('renders a primary outline button', () => {
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={ThemeBootstrap}>
                <Button variant="primary" outline>
                    Click Me!
                </Button>
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule(
            'background-color',
            ThemeBootstrap.Button.primaryOutline.backgroundColor
        );
        expect(tree).toMatchSnapshot();
    });

    it('renders a secondary outline button', () => {
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={ThemeBootstrap}>
                <Button variant="secondary" outline>
                    Click Me!
                </Button>
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule(
            'background-color',
            ThemeBootstrap.Button.secondaryOutline.backgroundColor
        );
        expect(tree).toMatchSnapshot();
    });
});
