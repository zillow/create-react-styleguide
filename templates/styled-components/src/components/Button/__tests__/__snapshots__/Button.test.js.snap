// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a primary button 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  background-color: #007bff;
  border-color: #007bff;
  color: #fff;
}

.c0:hover {
  background-color: #0069d9;
  border-color: #0062cc;
  color: #fff;
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button renders a primary outline button 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  background-color: transparent;
  border-color: #007bff;
  color: #007bff;
}

.c0:hover {
  background-color: #007bff;
  border-color: #007bff;
  color: #fff;
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button renders a secondary button 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  background-color: #6c747d;
  border-color: #6c757d;
  color: #fff;
}

.c0:hover {
  background-color: #5a6268;
  border-color: #545b62;
  color: #fff;
}

<button
  className="c0"
>
  Click Me!
</button>
`;

exports[`Button renders a secondary outline button 1`] = `
.c0 {
  cursor: pointer;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  -webkit-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out;
  background-color: transparent;
  border-color: #6c757d;
  color: #6c757d;
}

.c0:hover {
  background-color: #6c757d;
  border-color: #6c757d;
  color: #fff;
}

<button
  className="c0"
>
  Click Me!
</button>
`;
